### PR TITLE
Check gardenercluster deletion

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
 - implement two steps which are responsible for GardenerCluster deletion (deprovisioning process)
 
**Related issue(s)**
Related: https://github.com/kyma-project/kyma-environment-broker/issues/58

### CODE REVIEW

File: internal/process/steps/gardener_cluster.go [hunk 1]
```diff
@@ -53,13 +54,14 @@ func (s *checkGardenerCluster) Run(operation internal.Operation, log logrus.Fiel
 	state := gc.GetState()
 	log.Infof("GardenerCluster state: %s", state)
 	if state != GardenerClusterStateReady {
-		if time.Since(operation.UpdatedAt) > 15*time.Second {
+		// TODO: extract the timeout as a configuration setting, update the operation in the previous step to make the measurement correct way
+		if time.Since(operation.UpdatedAt) > 15*time.Minute {
```
Comment (piotrmiskiewicz) [inline internal/process/steps/gardener_cluster.go:58]: this timeout will be extracted as a config in a separate PR (as well as the saving operation to have a measurement)

File: internal/process/deprovisioning/check_gardenercluster_deleted_step.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func (step *CheckGardenerClusterDeletedStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if operation.KymaResourceNamespace == "" {
+		logger.Warnf("namespace for Kyma resource not specified")
+		return operation, 0, nil
+	}
+	resourceName := operation.GardenerClusterName
+	if resourceName == "" {
+		logger.Infof("GardenerCluster resource name is empty, skipping")
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply internal/process/deprovisioning/check_gardenercluster_deleted_step.go:43]: it can happen, if the provisioning failed before GardenerCluster creation

[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply internal/process/deprovisioning/check_gardenercluster_deleted_step.go:43]: I'm not 100% sure to have it as INFO, but for me it is the best level

File: internal/process/deprovisioning/check_gardenercluster_deleted_step.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	}
+}
+
+func (step *CheckGardenerClusterDeletedStep) Name() string {
+	return "Check_GardenerCluster_Deleted"
+}
+
+func (step *CheckGardenerClusterDeletedStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
+	if operation.KymaResourceNamespace == "" {
+		logger.Warnf("namespace for Kyma resource not specified")
```
Comment (jaroslaw-pieszka) [inline internal/process/deprovisioning/check_gardenercluster_deleted_step.go:38]: Why here is Warn and below Info - the result is the same, we skip the step.

Comment (jaroslaw-pieszka) [inline internal/process/deprovisioning/check_gardenercluster_deleted_step.go:38]: Here we skip as well. Why do not we log it as below.

Comment (piotrmiskiewicz) [reply internal/process/deprovisioning/check_gardenercluster_deleted_step.go:38]: namespace should be set before the step (somewhere in the provisioning initialisation)

File: internal/process/steps/lifecycle_manager_kubeconfig.go [hunk 4]
```diff
@@ -46,11 +46,7 @@ func (_ deleteKubeconfig) Name() string {
 func (s syncKubeconfig) Run(o internal.Operation, log logrus.FieldLogger) (internal.Operation, time.Duration, error) {
 	secret := initSecret(o)
 	if err := s.k8sClient.Create(context.Background(), secret); errors.IsAlreadyExists(err) {
-		if err := s.k8sClient.Update(context.Background(), secret); err != nil {
-			msg := fmt.Sprintf("failed to update kubeconfig secret %v/%v for lifecycle manager: %v", secret.Namespace, secret.Name, err)
-			log.Error(msg)
-			return s.operationManager.RetryOperation(o, msg, err, time.Minute, time.Minute*5, log)
-		}
```
Comment (piotrmiskiewicz) [inline internal/process/steps/lifecycle_manager_kubeconfig.go:53]: we should not update existing secret, because it is maintained by infra manager

**Review by jaroslaw-pieszka — APPROVED** [review]
